app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -78,9 +81,19 @@ def render_show_after_invalid
 
   def update_invalid_user
     current_user.second_factor_attempts_count += 1
-    # set time lock if max attempts reached
-    current_user.second_factor_locked_at = Time.zone.now if current_user.max_login_attempts?
-    current_user.save
+
+    UpdateUser.new(
+      user: current_user,
+      attributes: set_time_lock_if_max_attempts_reached
```

Comment:
eek ok this caused many spec failures because we need to increment user second factor attempts before checking `current_user.max_login_attempts?` (can't just set attr in a hash)

refactored / fixed at e9b5d9a